app/services/partner_routing_service.rb

```diff
@@ -9,11 +9,14 @@ def initialize(source_param: nil, zip_code: nil)
 
   # @return VitaPartner the object of the vita_partner we recommend routing to.
   def determine_partner
-    return vita_partner_from_source_param if @source_param.present? && vita_partner_from_source_param.present?
+    from_source_param = vita_partner_from_source_param if @source_param.present?
```

Comment:
I discovered while testing this that these methods were actually getting called 2x each, so I switched up the formatting. I think I could also memoize the results of each method in order to prevent this.